torch/ao/quantization/observer.py

```diff
@@ -1143,12 +1143,21 @@ def forward(self, x_orig: torch.Tensor) -> torch.Tensor:
         if x_orig.numel() == 0:
             return x_orig
         x = x_orig.detach()
+        x_min, x_max = torch.aminmax(x)
+        # want to ignore torch.inf since we don't actually
+        # want to make our quantization range infinite
+        # and in practice those values will be clamped
+        if x_min==-torch.inf or x_max==torch.inf:
```

Comment:
Thanks for the fix. Looks like the example in https://github.com/pytorch/pytorch/issues/100051 still fails. It may have case that `-3.4028234663852886 * 10**38` is not `-torch.inf`, but after it multiply by `self.upsample_rate` here https://github.com/pytorch/pytorch/blob/4c3799447f4cb8157e2145446c659b8629cb3c08/torch/ao/quantization/observer.py#L1095, it will change to `-torch.inf`.